lib/jekyll/document.rb

```diff
@@ -51,24 +52,28 @@ def data
     #
     # Returns the merged data.
     def merge_data!(other, source: "YAML front matter")
-      if other.key?('categories') && !other['categories'].nil?
-        if other['categories'].is_a?(String)
-          other['categories'] = other['categories'].split(" ").map(&:strip)
-        end
-        other['categories'] = (data['categories'] || []) | other['categories']
-      end
+      merge_categories!(other)
       Utils.deep_merge_hashes!(data, other)
-      if data.key?('date') && !data['date'].is_a?(Time)
-        data['date'] = Utils.parse_date(
-          data['date'].to_s,
+      if data.key?("date") && !data["date"].is_a?(Time)
+        data["date"] = Utils.parse_date(
+          data["date"].to_s,
           "Document '#{relative_path}' does not have a valid date in the #{source}."
         )
       end
       data
     end
 
+    def merge_categories!(other)
```

Comment:
Sorry, when you say each one gets an explicit private, you mean something like this:
```
private
def merge_categories!(other)
...
end

private
def merge_defaults
...
end
```
?
I've checked other files and it seems to be only one `private` keyword per file everywhere.
Just want to make sure I understand you correctly 😄